src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -1021,6 +1046,13 @@ public <T extends Serializable> T runRemotely(Step2<T> s) throws Throwable {
         }
     }
 
+    private HttpURLConnection decorateConnection(@NonNull URLConnection urlConnection) {
+        if (https && sslContext != null) {
+            ((HttpsURLConnection) urlConnection).setSSLSocketFactory(sslContext.getSocketFactory());
```

Comment:
If this is all we are using, does it make more sense to configure an optional https://docs.oracle.com/en/java/javase/17/docs/api/java.base/javax/net/ssl/SSLSocketFactory.html ?